tests/unit/plugins/modules/helper.py

```diff
@@ -76,12 +76,24 @@ def __init__(self, *args, **kwargs):
         self.mock_run_cmd = self._make_mock_run_cmd()
 
     def _make_mock_run_cmd(self):
-        call_results = [(x.rc, x.out, x.err) for x in self.run_cmd_calls]
-        error_call_results = (123,
-                              "OUT: testcase has not enough run_command calls",
-                              "ERR: testcase has not enough run_command calls")
+        def _results():
+            call_results = [(x.rc, x.out, x.err) for x in self.run_cmd_calls]
+            error_call_results = (123,
+                                  "OUT: testcase has not enough run_command calls",
+                                  "ERR: testcase has not enough run_command calls")
+            for result in chain(call_results, repeat(error_call_results)):
+                yield result
```

Comment:
Why not simply add a `raise Exception()` at the bottom which complains about `testcase has not enough run_command calls`, instead of inserting fake return values after `call_results` is empty? Like:
```suggestion
            call_results = [(x.rc, x.out, x.err) for x in self.run_cmd_calls]
            for result in call_results:
                yield result
            raise Exception("testcase has not enough run_command calls")
```

Reply:
I had an excellent reason for that when I wrote it, but it is under a pile of papers and excuses and I will never find it. :-P 
Thanks for the suggestion